docs/shared-content/scripts/create-script-step-scripts.include.md

```diff
@@ -0,0 +1,158 @@
+```powershell
+
+# Define working variables
+$octopusURL = "https://youroctourl"
+$octopusAPIKey = "API-YOURAPIKEY"
+$header = @{ "X-Octopus-ApiKey" = $octopusAPIKey }
+$space = "Spaces-1"
+
+# Project details
+$projectName = "MyProject"
+
+try
+{
+    # Get project
+    $project = (Invoke-RestMethod -Method Get -Uri "$octopusURL/api/$space/projects/all" -Headers $header) | Where-Object {$_.Name -eq $projectName}
+
+    # Get deployment process
+    $deploymentProcess = (Invoke-RestMethod -Method Get -Uri "$octopusURL/api/$space/deploymentprocesses/$($project.DeploymentProcessId)" -Headers $header)
+
+    # Get current steps
+    $steps = $deploymentProcess.Steps
+
+    # Add new step to process
+    $steps += @{
+      Name = "Run a script"
+      Properties = @{
+        'Octopus.Action.TargetRoles' = "Test-Role"
+      }  
+      Condition = "Success"
+      StartTrigger = "StartAfterPrevious"
+      PackageRequirement = "LetOctopusDecide"
+      Actions = @(
+        @{
+            ActionType = "Octopus.Script"
+            Name = "Run a script"
+            Environments = @()
+            ExcludedEnvironments = @()
+            Channels = @()
+            TenantTags = @()
+            Properties = @{
+                'Octopus.Action.RunOnServer' = "false"
+                'Octopus.Action.EnabledFeatures' = ""
+                'Octopus.Action.Script.ScriptSource' = "Inline"
+                'Octopus.Action.Script.Syntax' = "PowerShell"
+                'Octopus.Action.Script.ScriptFilename' = $null
+                'Octopus.Action.Script.ScriptBody' = "Write-Host `"Hello world`""
+            }
+            Packages = @()
+            IsDisabled = $false
+            WorkerPoolId = ""
+            WorkerPoolVariable = ""
+            Container = @{
+                "FeedId" = $null
+                "Image" = $null
+            }
+            CanBeUsedForProjectVersioning = $false
+            IsRequired = $false
+        }
+      )
+    }
+
+    # Convert steps to json
+    $deploymentProcess.Steps = $steps
+    $jsonPayload = $deploymentProcess | ConvertTo-Json -Depth 10
+
+    # Submit request
+    Invoke-RestMethod -Method Put -Uri "$octopusURL/api/$space/deploymentprocesses/$($project.DeploymentProcessId)" -Headers $header -Body $jsonPayload
+}
+catch
+{
+    Write-Host $_.Exception.Message
+}
+```
+```powershell PowerShell (Octopus.Client)
+Add-Type -Path 'Octopus.Client.dll' 
+
+$apikey = 'API-MCPLE1AQM2VKTRFDLIBMORQHBXA' # Get this from your profile
+$octopusURI = 'http://localhost' # Your server address
+
+$projectId = "Projects-100" # Get this from /api/projects
```

Comment:
This script uses `$projectId` but the first example uses the project name. Can we use name?